### PR TITLE
feature/60 allow per component node selector and tolerations overrides

### PR DESCRIPTION
**Pull request by omissis**
- feat: introduce per-component node selector and tolerations for the aws module
- feat: introduce per-component node selector and tolerations for the auth module
- feat: introduce per-component node selector and tolerations for the dr module
- feat: introduce per-component node selector and tolerations for the ingress module
- feat: introduce per-component node selector and tolerations for the logging and monitoring modules
- feat: introduce per-component node selector and tolerations for the policy module

### CODE REVIEW

**Review by nutellinoit — APPROVED** [review]
Overall LGTM, good job @omissis 

Just one question ☝️

File: templates/distribution/manifests/ingress/patches/infra-nodes.yml.tpl [hunk 1]
```diff
@@ -31,9 +14,9 @@ spec:
   template:
     spec:
       nodeSelector:
-        {{ template "nodeSelector" ( dict "spec" .spec ) }}
+        {{ template "nodeSelector" $cmArgs }}
```
[the remark this answers is not in the document]
Comment (omissis) [reply templates/distribution/manifests/ingress/patches/infra-nodes.yml.tpl:17]: will fix it, thanks!

**Review by nutellinoit — APPROVED** [review]
LGTM 🚀

File: pkg/schema/private/ekscluster_kfd_v1alpha2.go [hunk 2]
```diff
@@ -1316,1003 +1454,968 @@ func (j *SpecDistributionModulesIngressNginxTLSSecret) UnmarshalJSON(b []byte) e
...
 	}
-	if len(plain.CidrBlocks) < 1 {
-		return fmt.Errorf("field %s length: must be >= %d", "cidrBlocks", 1)
-	}
-	*j = SpecKubernetesNodePoolAdditionalFirewallRule(plain)
+	*j = SpecDistributionModulesIngress(plain)
 	return nil
 }
 
+type SpecDistributionModulesLoggingBanzai struct {
```
Comment (g-iannelli) [inline pkg/schema/private/ekscluster_kfd_v1alpha2.go:1625]:
```suggestion
type SpecDistributionModulesLoggingOperator struct {
```

File: pkg/schema/private/ekscluster_kfd_v1alpha2.go [hunk 3]
```diff
@@ -1316,1003 +1454,968 @@ func (j *SpecDistributionModulesIngressNginxTLSSecret) UnmarshalJSON(b []byte) e
...
-		return err
-	}
-	*j = SpecDistributionModulesIngressDNSPrivate(plain)
+	*j = SpecDistributionModulesLoggingOpensearch(plain)
 	return nil
 }
 
+type SpecDistributionModulesLogging struct {
+	// Banzai corresponds to the JSON schema field "banzai".
+	Banzai *SpecDistributionModulesLoggingBanzai `json:"banzai,omitempty" yaml:"banzai,omitempty"`
```
Comment (g-iannelli) [inline pkg/schema/private/ekscluster_kfd_v1alpha2.go:1732]:
```suggestion
type SpecDistributionModulesLogging struct {
	// Operator corresponds to the JSON schema field "operator".
	Banzai *SpecDistributionModulesLoggingOperator `json:"operator,omitempty" yaml:"operator,omitempty"`
```

File: pkg/schema/private/ekscluster_kfd_v1alpha2.go [hunk 4]
```diff
@@ -181,347 +389,408 @@ type SpecDistributionModulesAuthProviderBasicAuth struct {
...
+	return nil
 }
 
-type SpecDistributionModulesIngressNginxTLS struct {
-	// Provider corresponds to the JSON schema field "provider".
-	Provider SpecDistributionModulesIngressNginxTLSProvider `json:"provider" yaml:"provider"`
-
-	// Secret corresponds to the JSON schema field "secret".
-	Secret *SpecDistributionModulesIngressNginxTLSSecret `json:"secret,omitempty" yaml:"secret,omitempty"`
+type SpecDistributionModulesAwsSnapshotController struct {
```
Comment (g-iannelli) [inline pkg/schema/private/ekscluster_kfd_v1alpha2.go:637]:
```suggestion
type SpecDistributionModulesAwsEbsSnapshotController struct {
```

File: pkg/schema/private/ekscluster_kfd_v1alpha2.go [hunk 5]
```diff
@@ -181,347 +389,408 @@ type SpecDistributionModulesAuthProviderBasicAuth struct {
...
 
-	// Cert corresponds to the JSON schema field "cert".
-	Cert string `json:"cert" yaml:"cert"`
+	// Overrides corresponds to the JSON schema field "overrides".
+	Overrides TypesFuryModuleOverrides `json:"overrides" yaml:"overrides"`
 
-	// Key corresponds to the JSON schema field "key".
-	Key string `json:"key" yaml:"key"`
+	// SnapshotController corresponds to the JSON schema field "snapshotController".
+	SnapshotController *SpecDistributionModulesAwsSnapshotController `json:"snapshotController,omitempty" yaml:"snapshotController,omitempty"`
```
Comment (g-iannelli) [inline pkg/schema/private/ekscluster_kfd_v1alpha2.go:657]:
```suggestion
	SnapshotController *SpecDistributionModulesAwsEbsSnapshotController `json:"ebsSnapshotController,omitempty" yaml:"ebsSnapshotController,omitempty"`
```

**Review by g-iannelli — CHANGES_REQUESTED** [review]

**Review by g-iannelli — APPROVED** [review]
LGTM

**Pull request merged**